epochStart/bootstrap/storageProcess.go

```diff
@@ -236,26 +245,34 @@ func (sesb *storageEpochStartBootstrap) createStorageRequestHandler() (process.R
 	}
 
 	requestedItemsHandler := cache.NewTimeCache(timeBetweenRequests)
-	requestHandler, err := requestHandlers.NewResolverRequestHandler(
-		finder,
-		requestedItemsHandler,
-		sesb.whiteListHandler,
-		maxToRequest,
-		core.MetachainShardId,
-		timeBetweenRequests,
-	)
+	args := requestHandlers.ResolverRequestArgs{
+		RequestersFinder:      finder,
+		RequestedItemsHandler: requestedItemsHandler,
+		WhiteListHandler:      sesb.whiteListHandler,
+		MaxTxsToRequest:       maxToRequest,
+		ShardID:               core.MetachainShardId,
+		RequestInterval:       timeBetweenRequests,
+	}
+
+	factory, err := requestHandlers.NewResolverRequestHandlerFactory()
 	if err != nil {
 		return nil, err
 	}
 
 	switch sesb.chainRunType {
 	case common.ChainRunTypeRegular:
-		return requestHandler, nil
+		sesb.resolverRequestFactory = factory
 	case common.ChainRunTypeSovereign:
-		return requestHandlers.NewSovereignResolverRequestHandler(requestHandler)
+		sovFactory, sovErr := requestHandlers.NewSovereignResolverRequestHandlerFactory(factory)
+		if sovErr != nil {
+			return nil, sovErr
+		}
+		sesb.resolverRequestFactory = sovFactory
 	default:
 		return nil, fmt.Errorf("%w type %v", errors.ErrUnimplementedChainRunType, sesb.chainRunType)
 	}
+
+	return sesb.resolverRequestFactory.CreateResolverRequestHandler(args)
```

Comment:
I think this is a little bit overkill - we can go with it. But it is so much more code. If in the end we can create components with less code - if that is the end goal, than it is fine.

Reply:
the entire switch will be removed once the creator/factories will be injected and just the last return will be needed